plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,270 @@
+        private static final byte[] DECODE_MAP = new byte[ASCII_BITMASK + 1];
+
+        static {
+            Arrays.fill(DECODE_MAP, (byte) -1);
+            for (int i = 0; i < ENCODE_MAP.length; i++) {
+                DECODE_MAP[ENCODE_MAP[i]] = (byte) i;
+            }
+        }
+
+        private Base85Codec() {}
+
+        // This method will be used when supporting https://github.com/trinodb/trino/issues/17063
+        // This is used for testing codec round trip for now
+        public static String encodeBytes(byte[] input)
+        {
+            if (input.length % 4 == 0) {
+                return encodeBlocks(ByteBuffer.wrap(input));
+            }
+            int alignedLength = ((input.length + 4) / 4) * 4;
+            ByteBuffer buffer = ByteBuffer.allocate(alignedLength);
+            buffer.put(input);
+            while (buffer.hasRemaining()) {
+                buffer.put((byte) 0);
+            }
+            buffer.rewind();
+            return encodeBlocks(buffer);
+        }
+
+        private static String encodeBlocks(ByteBuffer buffer)
+        {
+            checkArgument(buffer.remaining() % 4 == 0);
+            int numBlocks = buffer.remaining() / 4;
+            // Every 4 byte block gets encoded into 5 bytes/chars
+            int outputLength = numBlocks * 5;
+            byte[] output = new byte[outputLength];
+            int outputIndex = 0;
+
+            while (buffer.hasRemaining()) {
+                long sum = buffer.getInt() & 0x00000000ffffffffL;
+                output[outputIndex] = ENCODE_MAP[(int) (sum / BASE_4TH_POWER)];
+                sum %= BASE_4TH_POWER;
+                output[outputIndex + 1] = ENCODE_MAP[(int) (sum / BASE_3RD_POWER)];
+                sum %= BASE_3RD_POWER;
+                output[outputIndex + 2] = ENCODE_MAP[(int) (sum / BASE_2ND_POWER)];
+                sum %= BASE_2ND_POWER;
+                output[outputIndex + 3] = ENCODE_MAP[(int) (sum / BASE)];
+                output[outputIndex + 4] = ENCODE_MAP[(int) (sum % BASE)];
+                outputIndex += 5;
+            }
+            return new String(output, US_ASCII);
+        }
+
+        public static ByteBuffer decodeBlocks(String encoded)
+        {
+            char[] input = encoded.toCharArray();
+            checkArgument(input.length % 5 == 0, "Input should be 5 character aligned");
+            ByteBuffer buffer = ByteBuffer.allocate(input.length / 5 * 4);
+
+            int inputIndex = 0;
+            while (buffer.hasRemaining()) {
+                long sum = 0;
+                sum += decodeInputChar(input[inputIndex]) * BASE_4TH_POWER;
+                sum += decodeInputChar(input[inputIndex + 1]) * BASE_3RD_POWER;
+                sum += decodeInputChar(input[inputIndex + 2]) * BASE_2ND_POWER;
+                sum += decodeInputChar(input[inputIndex + 3]) * BASE;
+                sum += decodeInputChar(input[inputIndex + 4]);
+                buffer.putInt((int) sum);
```

Comment:
resolve warning here. 
i guess we're consciously converting to an int value that's not numerically equivalent